examples/vnet/doca_gw_dpdk.c

```diff
@@ -845,56 +857,61 @@ doca_gw_dpdk_create_meter_rule(int port_id, uint32_t prof_id)
 	return meter_id++;
 }
 
-static int
-doca_gw_dpdk_destroy_meter_rule(int port_id, uint32_t mtr_id)
-{
-	struct rte_mtr_error error;
-
-	if (rte_mtr_destroy(port_id, mtr_id, &error)) {
-		DOCA_LOG_ERR("Port %u destroy meter(%d) error(%d) message: %s\n",
-			port_id, mtr_id, error.type,
-			error.message ? error.message : "(no stated reason)");
-		return -1;
-	}
-	return 0;
-}
-
 static int 
-doca_gw_dpdk_build_meter_action(struct doca_gw_pipelne_entry *pipe_entry, uint16_t port_id, 
-                                                        struct doca_dpdk_action_entry *entry, 
-							struct doca_gw_monitor *mon)
+doca_gw_dpdk_build_meter_action(struct doca_gw_pipelne_entry *pipe_entry,
+							uint16_t port_id,
+							struct doca_dpdk_action_entry *entry, 
+							struct doca_gw_monitor *mon,
+							struct doca_fwd_table_cfg *cfg)
 {
+	int ret;
 	struct rte_flow_action *action = entry->action;
 	struct rte_flow_action_meter *meter_conf;
 	struct doca_dpdk_action_meter_data *meter_data;
+	//uint32_t id = doca_id_pool_alloc_id(doca_gw_engine.meter_pool);  1 - 4096 
+	static uint32_t id = 0;
+
+	id++;
+	if (id <= 0) {
+		//TODO should be better handled
+		DOCA_LOG_ERR("out of meter profile ids");
+		return 0;
+	}
 
-	// todo: how to prevent profile create many times.
 	meter_data = &entry->action_data.meter;
-	meter_data->prof_id = doca_gw_dpdk_create_meter_profile(port_id, mon);
-	if (!meter_data->prof_id)
+	ret = doca_gw_dpdk_create_meter_profile(port_id, id, mon);
```

Comment:
This is a waste, why you need a profile per meter?
if user gives you cbs...etc on definition you can create single profile, if it fails you need to free it because it allocates resources on the pmd.